File: server/internal/models/todo.go
package models

import "time"

type Todo struct {
	ID          uint      `json:"id"`
    Content     string    `json:"content"`           
    Completed   bool      `json:"completed"`      
    CreatedAt   time.Time `json:"created_at"`
    UpdatedAt   time.Time `json:"updated_at"`  
}

type CreateTodoRequest struct {
    Content string `json:"content" binding:"required"`
}

type UpdateTodoRequest struct {
    Content   *string `json:"content"`    
    Completed *bool   `json:"completed"`  
}

File: web/src/types/todo.ts
export interface CreateTodoRequest {
    title: string;
    description: string;
}

export interface UpdateTodoRequest {
    completed: boolean;
}

// export interface TodoResponse {
//     id: string;
//     title: string;
//     description: string;
//     completed: boolean;
// }


export interface Todo {
    id: string;
    title: string;
    description: string;
    completed: boolean;
}

export type TodoFilter = 'all' | 'active' | 'completed';